baybe/surrogates/base.py

```diff
@@ -65,42 +80,103 @@ class Surrogate(ABC, SerialMixin):
     """Callable preparing surrogate inputs for training/prediction.
 
     Transforms a dataframe containing parameter configurations in experimental
-    representation to a corresponding dataframe containing their computational
-    representation. Only available after the surrogate has been fitted."""
+    representation to a corresponding dataframe containing their **scaled**
+    computational representation. Only available after the surrogate has been fitted."""
 
-    _target_transform: Callable[[pd.DataFrame], pd.DataFrame] | None = field(
+    _output_transform: Callable[[pd.DataFrame], pd.DataFrame] | None = field(
         init=False, default=None, eq=False
     )
-    """Callable preparing surrogate targets for training.
+    """Callable preparing surrogate outputs for training.
 
     Transforms a dataframe containing target measurements in experimental
-    representation to a corresponding dataframe containing their computational
-    representation. Only available after the surrogate has been fitted."""
+    representation to a corresponding dataframe containing their **scaled**
+    computational representation. Only available after the surrogate has been fitted."""
+
+    # TODO: type should be `Standardize | _NoTransform`` but is currently
+    #   omitted due to: https://github.com/python-attrs/cattrs/issues/531
+    _output_scaler = field(init=False, default=None, eq=False)
+    """Optional callable for scaling output values.
+
+    Scales a tensor containing target measurements in computational representation
+    to make them ready for processing by the surrogate. Only available after the
+    surrogate has been fitted."""
 
     def to_botorch(self) -> Model:
         """Create the botorch-ready representation of the model."""
         from baybe.surrogates._adapter import AdapterModel
 
         return AdapterModel(self)
 
-    def transform_inputs(self, data: pd.DataFrame) -> pd.DataFrame:
+    @staticmethod
+    def _make_parameter_scaler(
+        parameter: Parameter,
+    ) -> ParameterScalerProtocol | None:
+        """Return the scaler to be used for the given parameter."""
+        return MinMaxScaler()
+
+    @staticmethod
+    def _make_target_scaler() -> OutcomeTransform | None:
+        """Return the scaler to be used for target scaling."""
+        from botorch.models.transforms.outcome import Standardize
+
+        # TODO: Multi-target extension
+        return Standardize(1)
+
+    def _make_input_scaler(self, searchspace: SearchSpace) -> ColumnTransformer:
+        """Make the input scaler for transforming computational dataframes."""
+        from sklearn.compose import make_column_transformer
+
+        # Create the composite scaler from the parameter-wise scaler objects
+        transformers = [
+            (
+                "passthrough" if (s := self._make_parameter_scaler(p)) is None else s,
```

Comment:
I think you might be confusing what is the use of a `staticmethod` 🤔 It's not about whether called on instances or not but it defines if the raw function will be bound to an object (class/instance) that will be made accessible as the first argument. Since nothing needed, I make it a staticmethod. And the reason why I don't put it outside the class is because I want to have it in the same namespace, so that people can easily override it in their own subclasses.

If unclear, I suggest to dig for some resources. For example:
https://stackoverflow.com/questions/136097/what-is-the-difference-between-staticmethod-and-classmethod-in-python